File: code-smells-project/src/views/response.py
"""Canonical JSON response envelope helpers."""
from flask import jsonify


def success(data=None, message=None, http=200):
    payload = {"status": "ok", "sucesso": True}
    if data is not None:
        payload["dados"] = data
    if message is not None:
        payload["mensagem"] = message
    return jsonify(payload), http


def error(message, code="error", http=500):
    return jsonify({
        "status": "error",
        "sucesso": False,
        "erro": message,
        "code": code,
    }), http

File: ecommerce-api-legacy/src/views/response.js
'use strict';

function success(res, data, message, http = 200) {
  const payload = { status: 'ok' };
  if (data !== undefined) payload.data = data;
  if (message) payload.message = message;
  return res.status(http).json(payload);
}

function asyncHandler(fn) {
  return (req, res, next) => Promise.resolve(fn(req, res, next)).catch(next);
}

module.exports = { success, asyncHandler };
